### PR TITLE
Restrict collection slug to 30 char

### LINKED ISSUE #7487: Create collection form doesn't indicate that there is a 30 character limit for collection slug prior to submitting the form
### Describe the problem and steps to reproduce it:

* Go to https://addons.mozilla.org/en-US/firefox/collections/add/ and type in a collection name that is greater than 30 characters. 
* Click create collection 

### What happened?

I received an error saying "Ensure this field has no more than 30 characters."

### What did you expect to happen?

* help text near "collection ~~name~~" slug stating a character limit
* validation cues prior to clicking create collection
* ~~maybe make the "create collection" button insensitive to user input if more than 30 characters is entered~~ edit: we won't do that

### Anything else we should know?

**For QA:** I (@willdurand) believe the bug is in fact related to the `slug` field. When one creates a new collection, the collection name is used to create the slug. A collection name with more than 30 chars is allowed but not a slug. When the slug is auto-generated from the name, we should limit this generated value to 30 chars.

### PR DESCRIPTION
**Pull request by championshuttler**
Fixes #7487 

After:
image

### CODE REVIEW

**Comment by championshuttler** [comment]
@willdurand i guess this is enough to add after https://github.com/mozilla/addons-frontend/blob/master/tests/unit/amo/components/TestCollectionManager.js#L183
```
    expect(root.find('#collectionName')).toHaveProp('maxLength', '30');
```

Let me know if I am doing some mistakes. Thanks

**Comment by championshuttler** [comment]
@willdurand so by adding `maxLength` in `slug` input , its not helping to restrict the slug to 30 chars when we input the name , but when we input the slug manually its working. Can you help me with that?

Thanks

**Comment by championshuttler** [comment]
@willdurand so after adding `substring` next to `.join` its working the way we expecting i guess, it only takes <=30 char from the collection name. Thanks to rebm.

Ready for another review.

File: src/amo/constants.js [hunk 1]
```diff
@@ -23,3 +23,6 @@ export const LANDING_PAGE_EXTENSION_COUNT = 4;
 export const LANDING_PAGE_THEME_COUNT = 3;
 
 export const DOWNLOAD_FIREFOX_BASE_URL = 'https://www.mozilla.org/firefox/new/';
+
+// Length of the collection slug
```
Comment (championshuttler) [inline src/amo/constants.js]: Can you suggest something better to me 👻

Comment (willdurand) [reply src/amo/constants.js]: What do you mean?

File: src/amo/components/CollectionManager/index.js [hunk 2]
```diff
@@ -288,6 +290,7 @@ export class CollectionManagerBase extends React.Component<
           <input
             onChange={this.onTextInput}
             id="collectionSlug"
+            maxLength={SLUG_MAX_LENGTH}
```
Comment (willdurand) [inline src/amo/components/CollectionManager/index.js]: I'd suggest moving the constant `SLUG_MAX_LENGTH` in this file instead of `amo/constants`.

Comment (championshuttler) [reply src/amo/components/CollectionManager/index.js]: Done 😄

File: src/amo/components/CollectionManager/index.js [hunk 3]
```diff
@@ -184,6 +183,8 @@ export class CollectionManagerBase extends React.Component<
 
     const trimmedValue = value.trim();
 
+    const SLUG_MAX_LENGTH = 30;
```
Comment (willdurand) [inline src/amo/components/CollectionManager/index.js]: If you put it outside of the class, then you can `export` it and re-`import` in your test case.

Comment (championshuttler) [reply src/amo/components/CollectionManager/index.js]: I was thinking so , but I thought that we are using it only in one assertion so.. . 

Making the changes. Thanks

**Comment by willdurand** [comment]
Thanks @championshuttler!

**Pull request merged**